example/src/screens/MainScreen.tsx

```diff
@@ -74,11 +99,76 @@ export default class MainScreen extends Component<any, MainScreenState> {
       .then((response) => response.data);
   }
 
+  componentDidMount() {
+    this.updateTrackingConsent()
+  }
+
+  updateTrackingConsent() {
+    getTrackingConsent().then(consent => {
+      this.setState({
+        trackingConsent: consent
+      })
+    })
+  }
+
+  setTrackingConsentModalVisible(visible: boolean) {
+    this.setState({ trackingConsentModalVisible: visible })
+  }
+
+  renderTrackingConsentItem({ item }) {
+    const isSelected = TrackingConsent[item] === this.state.trackingConsent
+    const backgroundColor = isSelected ? "#303f9f" : "#448aff";
+    const color = isSelected ? 'white' : 'black';
+
+    return (
+      <TrackingConsentItem
+        item={item}
+        onPress={() => {
+          const consent = TrackingConsent[item]
+          this.setState({ trackingConsent: consent })
+        }}
+        backgroundColor={{ backgroundColor }}
+        textColor={{ color }}
+      />
+    );
+  }
+
   render() {
     return <View style={styles.defaultScreen}>
       <Text>{this.state.welcomeMessage}</Text>
       <View style={{ marginTop: 40, alignItems: "center" }}>
-        <Text>{this.state.callDatadogButtonAction}</Text>
+        <Button
+          title={`Tracking Consent: ${this.state.trackingConsent}`}
+          accessibilityLabel="update_tracking_consent"
+          onPress={() => this.setTrackingConsentModalVisible(true)}
+        />
+        <Modal
```

Comment:
I think we can extract this in a separated component. You can see more examples in the `shopist-reactnative` application: https://github.com/DataDog/shopist-reactnative/blob/master/app/components/cart/CartListFooter.js

Reply:
thanks! I wanted to avoid having multiple components in order to keep it simple (no need to pass the data between), but this example can help then.